Dockerfile

```diff
@@ -1,10 +1,39 @@
-FROM golang
+# Build the spiffe-helper binary
+FROM --platform=$BUILDPLATFORM golang:1.21.5-alpine as base
+WORKDIR /workspace
 
-WORKDIR /build
-COPY . /build
-RUN CGO_ENABLED=0 go build -o spiffe-helper ./cmd/spiffe-helper
+# Cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+COPY go.* ./
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
+
+# Copy the go source
+COPY cmd/spiffe-helper/main.go cmd/spiffe-helper/main.go
+COPY pkg/ pkg/
+
+# xx is a helper for cross-compilation
+# when bumping to a new version analyze the new version for security issues
+# then use crane to lookup the digest of that version so we are immutable
+# crane digest tonistiigi/xx:1.3.0
+FROM --platform=${BUILDPLATFORM} tonistiigi/xx@sha256:904fe94f236d36d65aeb5a2462f88f2c537b8360475f6342e7599194f291fb7e AS xx
+
+FROM --platform=${BUILDPLATFORM} base as builder
+ARG TARGETPLATFORM
+ARG TARGETARCH
+
+ENV CGO_ENABLED=0
+COPY --link --from=xx / /
+RUN xx-go --wrap
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    go build -o bin/spiffe-helper cmd/spiffe-helper/main.go
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+#FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static AS spiffe-helper
+WORKDIR /
+COPY --link --from=builder /workspace/bin/spiffe-helper /spiffe-helper
 
-FROM scratch
-COPY --from=0 /build/spiffe-helper /spiffe-helper
 ENTRYPOINT ["/spiffe-helper"]
-CMD ["-config", "/etc/spiffe-helper.conf"]
+CMD []
```

Comment:
There a reason to switch this? Not a big deal, but for things using the container, they don't need to specify any args if they mount in the config in the right place, making it a bit easier to use. For those that don't want to use the config file, they just override CMD as normal?

So, docker run -it --rm spiffe-helper --help   works as expected.

Reply:
This kind of matches what spire is doing:

```
$ podman run -it --rm ghcr.io/spiffe/spire-agent:1.8.5
could not find config file /opt/spire/conf/agent/agent.conf: please use the -config flag
```

It assumes config file exists if you don't override any flags.

Reply:
spiffe helper assumes a default config file too:

```
/spiffe-helper --help
Usage of ./spiffe-helper:
  -config string
    	<configFile> Configuration file path (default "helper.conf")
```

Reply:
helper.conf in the current directory, or helper.conf in the same dir as the helper binary?

Reply:
same dir as binary.

Reply:
In the case of spire-agent, the place it looks is in a config directory, not in the bin directory, so its pretty easy to mount a configmap into the place its looking by default. In the case of the spire-helper, the default is in the bindir then, so can't mount it there easily without doing subdir mounts and putting a config file in a bindir feels a bit weird.

So the default is a bit harder to use in a containerized environment then it is with spire-agent. Maybe we could consider changing the default to match more what spire-agent is doing. but probably not right to do in this pr.

Reply:
Yes, i think we should update the default. Opened an issue for this: https://github.com/spiffe/spiffe-helper/issues/129